### PR TITLE
Replace fmt stub with real implementation

### LINKED ISSUE #7634: Bring in fmt to replace our stub
**folly** has a dependency on **fmt**.  We should bring in **fmt** before we hit some blocking issue that requires us to do it in a rush.

Thus far we have gotten away with stubbing out places where this is an issue, but usage within **folly** is expanding, and its only a matter for time before we hit a case where we need the real implementation.

**fmt** is already used by other Microsoft projects, such as **Windows Terminal**, so any issues related to building **fmt** with MSVC etc are not increases in total work needed to support fmt across the company.

### PR DESCRIPTION
**Pull request by acoates-ms**
Fixes #7634 

Basically mirrored off how we do folly today.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7635)

### CODE REVIEW

File: vnext/PropertySheets/Bundle.Common.targets [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
   <!-- See https://github.com/facebook/react-native/blob/07d090dbc6c46b8f3760dbd25dbe0540c18cb3f3/react.gradle#L190 for reference -->
   <Target Name="CompileHermesBytecode" AfterTargets="MakeBundle" Condition="'$(UseBundle)' == 'true' AND '$(UseHermes)' == 'true'">
+    <Message Importance="High" Text="Running [$(HermesCompilerCommand) -emit-binary -out $(BundleOutputFile).hbc $(BundleOutputFile) $(HermesCompilerFlags) -output-source-map] to precompile bundle file." />
```
Comment (asklar) [inline vnext/PropertySheets/Bundle.Common.targets:21]: unrelated?

Comment (NickGerleman) [reply vnext/PropertySheets/Bundle.Common.targets:21]: Seems like a good change though

File: vnext/fmt/fmt.vcxproj [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      <IgnoreAllDefaultLibraries>false</IgnoreAllDefaultLibraries>
+      <GenerateWindowsMetadata>false</GenerateWindowsMetadata>
+    </Link>
+  </ItemDefinitionGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <Target Name="DownloadFmt" BeforeTargets="PrepareForBuild">
+    <Message Importance="High" Text="Downloading fmt..." Condition="!Exists('$(FmtDir)..\.fmtzip\fmt-$(FmtVersion).zip')" />
+    <DownloadFile Condition="!Exists('$(FmtDir)..\.fmtzip\fmt-$(FmtVersion).zip')" SourceUrl="https://github.com/fmtlib/fmt/archive/refs/tags/$(FmtVersion).zip" DestinationFileName="fmt-$(FmtVersion).zip" DestinationFolder="$(FmtDir)..\.fmtzip" Retries="10" />
+  </Target>
+  <Target Name="UnzipFmt" BeforeTargets="PrepareForBuild" DependsOnTargets="DownloadFmt">
```
Comment (NickGerleman) [inline vnext/fmt/fmt.vcxproj:114]: Probably not worth the effort yet, but if we see more of these in the future, it might be nice to create a shared MSBuild task for "Download GitHub archive dependency into node_modules".

**Review by NickGerleman — APPROVED** [review]

**Comment by asklar** [comment]
@acoates-ms Please make sure we *do not* bring down/bring along the doc/ folder as that has a different license

File: vnext/PropertySheets/Bundle.Common.targets (hunk 1 above)
Comment (asklar) [reply vnext/PropertySheets/Bundle.Common.targets:21]: yes, can we put this in a separate change though?

**Comment by asklar** [comment]
Does any of fmt need to go into any of the nuget packages nuspecs?

**Comment by acoates-ms** [comment]
The only thing in the npm package will be the project files.

**Review by asklar — CHANGES_REQUESTED** [review]
setting request changes to make sure we're ok with the doc folder licensing and the nuget nuspecs (feel free to clear me once you see this :))

**Comment by acoates-ms** [comment]
As far as I know, fmt is internal to the dll, so shouldn't need to be in the nuspec.

**Review by asklar — APPROVED** [review]

**Review by dannyvv — APPROVED** [review]
